### PR TITLE
[Issue #240] bugfix

### PR DESCRIPTION
**Pull request by gsmol**
When remote agent encounters an error condition during exectuion of async task, save the error message and return it when requested. Also handle the SIGPIPE signal.

### CODE REVIEW

File: src/utils/file.c [hunk 1]
```diff
@@ -1094,6 +1306,32 @@ typedef struct fioGZFile
 	Bytef    buf[ZLIB_BUFFER_SIZE];
 } fioGZFile;
 
+/* check if remote agent encountered any error during execution of async operations */
+int
+fio_check_error_fd_gz(gzFile f, char **errmsg)
+{
+	if (f && ((size_t)f & FIO_GZ_REMOTE_MARKER))
```
Comment (alubennikova) [inline src/utils/file.c:1313]: Could you please add a comment to FIO_GZ_REMOTE_MARKER definition?

File: src/utils/file.c [hunk 2]
```diff
@@ -747,20 +855,124 @@ ssize_t fio_fwrite_compressed(FILE* f, void const* buf, size_t size, int compres
...
+
+		/* send header */
+		IO_CHECK(fio_write_all(out, &hdr, sizeof(hdr)), sizeof(hdr));
+
+		/* send message itself */
+		IO_CHECK(fio_write_all(out, async_errormsg, hdr.size), hdr.size);
+
+		//TODO: should we reset the tainted state ?
+//		pg_free(async_errormsg);
+//		async_errormsg = NULL;
```
Comment (alubennikova) [inline src/utils/file.c:968]: I don't see, where we reset async_errormsg.
How would we handle several consecutive errors?

Comment (gsmol) [reply src/utils/file.c:968]: We don`t. Currently it is assumed, that a first error is the last error.

File: src/utils/file.c (hunk 1 above)
Comment (gsmol) [reply src/utils/file.c:1313]: Sure

**Comment by gsmol** [comment]
Merged

**Pull request closed without merging**